### PR TITLE
Milestone #1

### PR DESCRIPTION
**Pull request by simandebvu**
## Added 

- HTML sizes for sm, md and lg screen types
- Relevant padding and margin
- Anchor tags for elements in page
- Screenshots for elements
- HTML and CSS Validation (Stickler)
- Site Screenshot encompassing device types

### CODE REVIEW

**Comment by rammazzoti2000** [comment]
## Pull Request under review by Alexandru Bangau (TSE)

File: assets/css/style.css [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+/* NAVIGATION */
+.navbar {
+  background-color: var(--main-page-color);
+}
+
+.navbar-brand img {
+  height: 20px;
+  padding-left: 3px;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `padding-left` css rule can be used as bootstrap class --> `pl-2`

File: assets/css/style.css [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.nav-button {
+  font-size: 14px;
+  text-transform: uppercase;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `text-transform` css rule can be used as bootstrap class --> `text-uppercase` (simple as that :smiley: )

File: assets/css/style.css [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.navbar-light .navbar-nav .nav-link {
+  color: rgba(0, 0, 0);
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `color` (black) css rule can be used as a bootstrap class --> `text-dark`

File: assets/css/style.css [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.navbar-light .navbar-nav .nav-link:focus,
+.navbar-light .navbar-nav .nav-link:hover {
+  color: var(--main-page-color);
+}
+
+.secondary-nav {
+  background-color: #fff !important;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: no use of `!important` properties!
`background-color` (white) class selector can be used as a bootstrap class --> `bg-white`

File: assets/css/style.css [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.nav-item a {
+  font-size: 12px;
+  line-height: 20px;
+  font-weight: bold;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `font-weight` rule css can be used as bootstrap class --> `font-weight-bold`

File: assets/css/style.css [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+  font-family: "Roboto Condensed", sans-serif;
+  text-transform: capitalize;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `text-transform`  css rule can be used as bootstrap class --> `text-capitalize`

File: assets/css/style.css [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.login-button {
+  color: #fff;
+  text-decoration: none;
+  background-color: transparent;
+  margin-right: 2px;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: all these css rules can be used as bootstrap classes:
`text-white text-decoration-none bg-transparent mr-1`

File: assets/css/style.css [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.sub-button {
+  color: var(--main-page-color);
+  background-color: #fff;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `background-color` css rule can be used as bootstrap class --> `bg-white`

File: assets/css/style.css [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+  border-radius: 0;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `border-radius` css rule can be used as bootstrap class --> `rounded-0`

File: assets/css/style.css [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+/* HELPER CLASSES */
+.link-deco,
+.link-no-deco,
+.article-link {
+  color: #000;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `color` (black) css rule can be used as bootstrap class --> `text-dark`

File: assets/css/style.css [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.link-deco:hover {
+  color: var(--main-page-color);
+  text-decoration: none;
+}
+
+.bold-text {
+  font-weight: bold;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: font-weight rule css can be used as bootstrap class --> font-weight-bold

File: assets/css/style.css [hunk 12]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+p {
+  padding-top: 10px;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: `padding-top` css rule can be used as bootstrap class --> `pt-3`

File: assets/css/style.css [hunk 13]
```diff
@@ -0,0 +1,379 @@
...
+.magazine-header {
+  display: flex;
+  justify-content: center;
+  font-weight: bold;
+}
+
+.opinion-item {
+  display: flex;
+  align-items: center;
+  padding: 5px;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: all these css rules can be used as bootstrap classes:
`d-flex align-items-center p-2`

File: assets/css/style.css [hunk 14]
```diff
@@ -0,0 +1,379 @@
...
+
+.opinion-item img {
+  margin-right: 5px;
+}
+
+.debate-side {
+  padding-top: 10px;
+  padding-bottom: 4px;
+  display: flex;
+  align-items: center;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: same as above

File: assets/css/style.css [hunk 15]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+.magazine-images {
+  position: relative;
+  text-align: center;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: these css rules can be used as bootstrap classes
`position-relative text-center`

File: assets/css/style.css [hunk 16]
```diff
@@ -0,0 +1,379 @@
...
+.magazine-images img {
+  height: 228px;
+  width: 170px;
+}
+
+.magazine-info {
+  font-weight: bold;
+  position: relative;
+  display: flex;
+  flex-direction: column;
```
Comment (rammazzoti2000) [inline assets/css/style.css]: these classes can be used as bootstrap classes
`d-flex flex-column position-relative font-weight-bold`

**Review by rammazzoti2000 — CHANGES_REQUESTED** [review]
### Hello @simandebvu I'm Alex your code reviewer for this project :raised_hands: 
Good attempt this far on this project, I see a descriptive Readme file, Pull Request summary descriptive as well, the commits could've been made more often though (remember small commits with meaningful description), also good use on bootstrap classes but I think you can do better than this :muscle: 
So as you expected already, there are still issues to fix for this project before it cam be approved, follow the comments bellow :point_down: 
## :stop_sign: Required Changes :construction: :construction_worker: 
- [x] More use of bootstrap classes, remember this is a bootstrap project, see the inline comments :point_down: 
- [x] Overall the page needs more work on typography, colors and alignment
###
### On Large Screens (Desktop) :desktop_computer: 
- [x] The navigation bar is not as the original and the logo is too big, also the links are a bit off, should be center aligned, and you are also missing the weather widget
### Clone:
image
### Original:
image
####
- [x] Overall the page is missing some space on left and right (meaning is taking all the width when it should not)
### Clone:
image
### Original:
image
 - [x] In the `Editor's Pick` section you are missing a row with four elements, the original has two rows with four elemenst each
image
- [x] Your `Subscribe` section is a bit off, it should look like the original
### Clone:
image
### Original:
image
- [x] The footer is not like original, please check the screenshots
### Clone:
image
### Original:
image
### On Medium Screens (Tablet) 
- [x] The `Featured Stories` section is missing some content, please add at least 3 more articles to it
- [x] The `Sign Up` feature should be like the original
### Clone:
image

### Original:
image
- [x] The `U.S., Sport, Business, etc. ` categories have the vertical alignment a bit off, you can notice the left side is not aligned with the right side, also the right side has an extra top border that shouldn't be there
image
- [x] The `Subscribe`  & `News Letter` should look like the original
### `Subscribe` - Original:
image
### `Newsletter` - Original:
image
- [x] The footer should look like the original
### Original:
image
### Small Screens (Smartphones) :iphone: 
- [x] The order of the top main content is wrong. While yours is `top, heroes of pandemic, culture, more stories, featured stories, the debate, opinion`, the original is `top stories, featured stories, the debate, opinion, more stories, heroes of pandemic, culture and travel`
- [x] The `Signup` feature keeps being a bit off
### Clone:
image
### Original:
image
- [x] The subsections of each dedicated category must be properly aligned along with the top content
image
- [x] The `Subscribe`, `Newsletter` are still broken and are not looking like the original
### `Subscribe` - Original:
image
### `Newsletter` - Original:
image

- [x] Footer needs more work as well, it should look like the original
image
### <em>Kindly fix the REQUIRED issues and send another code review request.

Cheers and Happy coding! :clap: 

Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of alexandru if you have any additional question)</em>
### :stop_sign: Status: Required Changes

File: assets/css/style.css (hunk 13 above)
Comment (simandebvu) [reply assets/css/style.css]: Thank you so so much for this! Am much much wiser now 😄

**Comment by simandebvu** [comment]
## Changes made 

- Navbar with a weather widget and padding on large screens
- Changed font family and size for the navbar
- Subscribe, newsletter and news sections
- Modified footer with columns
- Added more content to the features section
- Removed code that is inherent in bootstrap
- Validated with W3c validator and stylelint
- Modified Signup feature for mobile
- Added image on the newsletter to show on large screens

**Comment by santiago-dev96** [comment]
# Pull Request under review by @santiago-rodrig (TSE)

**Review by santiago-dev96 — CHANGES_REQUESTED** [review]
# Changes required :no_entry_sign:

You are getting closer, but there are some **details** that need to be resolved.

1. Each section must have a corresponding heading

Screen Shot 2020-05-25 at 9 54 32 AM

2. The "More Stories" section needs a bit more separation between its items.

Screen Shot 2020-05-25 at 9 54 41 AM

3. The section with categories doesn't display each category as in the original page.

**Original**

Screen Shot 2020-05-25 at 9 57 17 AM

**Yours**

Screen Shot 2020-05-25 at 9 57 08 AM

Please submit another CRR after implementing the changes.

You can find me in slack as **srodrig**, have a nice day! :wave:

**Comment by simandebvu** [comment]
## Changes done 😄 

- More stories and page list item improvements
- Added titles to sections and removed all uppercase titles so it looks more like the original
- Spacing between items in the __More__ section

**Comment by addod19** [comment]
## Pull Request under review by Daniel Larbi Addo (TSE)

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @simandebvu, great work with this amazing project :+1: You did a great job fixing most of the comments pointed out by the previous TSE's :slightly_smiling_face: 
Let us address the comment below to make our project even better:
- The second image in the ```More stories``` is not displaying fully, kindly fix it
mmore

Kindly implement this change and submit another code review :smile:
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by simandebvu** [comment]
## Changes made 😄 

- Removed the scrolling effect from the __More Stories__ Section

**Review by christianotieno — CHANGES_REQUESTED** [review]
# CHANGES REQUIRED :warning:
Hey @simandebvu   :wave:
-
You have done a splendid job! :tada: Keep it up.
We, however, do have some few changes to carry out before we proceed (see review comment).
##
## REQUIRED CHANGES
### LARGE SCREEN CHANGES: 
- [x] The general layout of the page is not properly centred as with the source page. The [`container`](https://www.w3schools.com/bootstrap4/bootstrap_containers.asp) class would be appropriate in the centring of the whole page content. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|newsweek|newsweek-clone|
##
- [x] In the magazine section is not properly styled as with the source page.. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|in-the-magazine|in-the-magazine-clone|
##
- [x] Editor's pick section. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|editors-pick|editors-pick-clone|
##
- [x] `US`, `World`, `Business`, `Tech & Science`, and `Sports`'s section See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|US-section|US-section-clone|

- [x] The `footer` and the `subscribe` section. See the difference below 👇 

|[Source](https://www.newsweek.com/)|[Clone](https://raw.githack.com/simandebvu/newsweek-clone/develop/index.html)|
|:---:|:---:|
|foooter|foooter-clone|

##
- [x] Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)).

- [x] Make sure that basic CSS bad practices are not used ([article for support ](https://speckyboy.com/good-bad-css-practices)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright).

### MEDIUM & SMALL SCREEN CHANGES: 
-  [x] Conduct the same changes in the medium and small screen changes as well. 
- The goals here will be to get the clone to appear strikingly similar to the source page. Having assessed your skills I believe you are capable of cons=duciting these changes.  Happy coding.

 ## OPTIONAL
- [x] Validate your CSS using this [tool](https://jigsaw.w3.org/css-validator/validator).

- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Comment by simandebvu** [comment]
## Changes made

- Added removed container-fluid and added a container element to the body
- Edited the **in the magazine** and **editors picks** to look more like the original
- Resolved the **subscribe part** not rendering properly in Full HD Screens
- Newsletter image improvements
- CSS and HTML validation

**Review by christianotieno — CHANGES_REQUESTED** [review]
Hello @simandebvu 👋 
You have done a great job with the changes. Just some few more changes and we should be done here.
## Required
- During the screen breakpoints, the contents in both the medium and small screens should cover the whole width of the screen. The `container` class is only appropriate for large screens only. Kindly fix this issue then submit for another code review. Thank you

**Comment by simandebvu** [comment]
## Changed

- Fullscreen for medium and small screens

File: assets/css/style.css [hunk 17]
```diff
@@ -0,0 +1,375 @@
...
+
+  .newsletter-thumb {
+    display: none;
+  }
+}
+
+@media screen and (min-width: 992px) {
+  .nav-button {
+    font-size: 17px;
+  }
```
Comment (trekab) [inline assets/css/style.css]: This is a great place to rely on bootstrap classes. [Responsive breakpoints](https://getbootstrap.com/docs/4.3/layout/overview/#responsive-breakpoints)

**Review by trekab — CHANGES_REQUESTED** [review]
# Changes Required! :recycle: 
Hi @simandebvu, you have done a good job implementing the feedback from the T.S.E's. However, there are a few changes to make.
- [x] For the `Featured slideshows` section, your clone looks different from the original on medium screens.
__Clone__:
Clone
__Source__:
Source
- [x] Using Bootstrap Css classes **as mauch as possible** for styling (for example [Grid layout](https://getbootstrap.com/docs/4.3/layout/overview/) and [Flex](https://getbootstrap.com/docs/4.3/layout/utilities-for-layout/#flexbox-options)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/using_bootstrap/bootstrap_classes_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540308/)) is ideal for this project. Check the comment below for an example of where you could use bootstrap.

__Please make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__

**Comment by simandebvu** [comment]
## Changes made

- Updated featured slide show section on md devices
- Added more bootstrap classes

**Review by christianotieno — APPROVED** [review]
# PROJECT APPROVED :champagne:
Excellent job with the changes @simandebvu  :fireworks: 
##
***All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.***
- Congratulations on the splendid job.
- You can now merge to master. I wish you all the very best in your next tasks 🥇 
##
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)

**Pull request merged**